lib/backend/src/Obelisk/Backend.hs

```diff
@@ -193,16 +208,37 @@ runBackend backend frontend = case checkEncoder $ _backend_routeEncoder backend
 renderGhcjsFrontend
   :: (MonadSnap m, HasCookies m)
   => (route -> Text)
-  -> Text -- ^ URL to GHCJS app JavaScript
+  -> Text
+  -> (Text -> FrontendWidgetT route ()) -- ^ GHCJS all.js script widget
```

Comment:
Maybe it makes more sense to not pass both `allJsUrl` and this script everywhere. Why not "close over" the URL and pass just the script?

Reply:
We use the path in the preload widget, which isn't customisable here. I thought about introducing a type to wrap this up, but didn't come to anything that felt "right".

Reply:
Fair.

Reply:
I wonder if instead we should allow customising both preload widget and script widget? That shifts the problem slightly while giving the user a little more control.

Reply:
The two widgets would fit fairly sensibly in a config record, I think.

Reply:
I like it.